## Import Libaries

In [1]:
import pandas as pd
import random

## Load dataset

In [2]:
!ls

README.md
SSIS
dataset
preprocess_data.ipynb
report


In [3]:
!ls dataset

marketing_data.csv


In [4]:
df = pd.read_csv('dataset/marketing_data.csv')
df.head(5)

ID  Year_Birth   Education Marital_Status      Income   Kidhome  \
0   1826        1970  Graduation       Divorced  $84,835.00         0   
1      1        1961  Graduation         Single  $57,091.00         0   
2  10476        1958  Graduation        Married  $67,267.00         0   
3   1386        1967  Graduation       Together  $32,474.00         1   
4   5371        1989  Graduation         Single  $21,474.00         1   

   Teenhome Dt_Customer  Recency  MntWines  ...  NumStorePurchases  \
0         0   6/16/2014        0       189  ...                  6   
1         0   6/15/2014        0       464  ...                  7   
2         1   5/13/2014        0       134  ...                  5   
3         1   5/11/2014        0        10  ...                  2   
4         0    4/8/2014        0         6  ...                  2   

   NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  \
0                  1             0             0             0             0   
1                  5             0             0             0             0   
2                  2             0             0             0             0   
3                  7             0             0             0             0   
4                  7             1             0             0             0   

   AcceptedCmp2  Response  Complain  Country  
0             0         1         0       SP  
1             1         1         0       CA  
2             0         0         0       US  
3             0         0         0      AUS  
4             0         1         0       SP  

[5 rows x 28 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 28 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   ID                   2240 non-null   int64 
 1   Year_Birth           2240 non-null   int64 
 2   Education            2240 non-null   object
 3   Marital_Status       2240 non-null   object
 4    Income              2216 non-null   object
 5   Kidhome              2240 non-null   int64 
 6   Teenhome             2240 non-null   int64 
 7   Dt_Customer          2240 non-null   object
 8   Recency              2240 non-null   int64 
 9   MntWines             2240 non-null   int64 
 10  MntFruits            2240 non-null   int64 
 11  MntMeatProducts      2240 non-null   int64 
 12  MntFishProducts      2240 non-null   int64 
 13  MntSweetProducts     2240 non-null   int64 
 14  MntGoldProds         2240 non-null   int64 
 15  NumDealsPurchases    2240 non-null   int64 
 16  NumWeb

## Rename Column

In [6]:
df.rename(columns={'ID':'User_ID'}, inplace=True)
df.rename(columns={' Income ':'Income'}, inplace=True)
df.rename(columns={'Dt_Customer': 'Date_Enroll'}, inplace = True)
df.rename(columns={'MntMeatProducts': 'MntMeats'}, inplace = True)
df.rename(columns={'MntFishProducts': 'MntFishs'}, inplace = True)
df.rename(columns={'MntSweetProducts': 'MntSweets'}, inplace = True)
df.rename(columns={'MntGoldProds': 'MntGolds'}, inplace = True)

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 28 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   User_ID              2240 non-null   int64 
 1   Year_Birth           2240 non-null   int64 
 2   Education            2240 non-null   object
 3   Marital_Status       2240 non-null   object
 4   Income               2216 non-null   object
 5   Kidhome              2240 non-null   int64 
 6   Teenhome             2240 non-null   int64 
 7   Date_Enroll          2240 non-null   object
 8   Recency              2240 non-null   int64 
 9   MntWines             2240 non-null   int64 
 10  MntFruits            2240 non-null   int64 
 11  MntMeats             2240 non-null   int64 
 12  MntFishs             2240 non-null   int64 
 13  MntSweets            2240 non-null   int64 
 14  MntGolds             2240 non-null   int64 
 15  NumDealsPurchases    2240 non-null   int64 
 16  NumWeb

## Categorical value column Country

In [8]:
df['Country'].unique()

array(['SP', 'CA', 'US', 'AUS', 'GER', 'IND', 'SA', 'ME'], dtype=object)

In [9]:
value_to_replace = {'SA': 'South Africa', 
                    'CA': 'Canada', 
                    'SP': 'Spain', 
                    'GER': 'Germany', 
                    'IND': 'India', 
                    'US': 'United States',
                    'AUS': 'Australia',
                   'ME': 'Montenegro'}
df['Country'].replace(value_to_replace, inplace=True)

In [10]:
df['Country'].head(5)

0            Spain
1           Canada
2    United States
3        Australia
4            Spain
Name: Country, dtype: object

## Format datatime
Ta cần chuyển cột **Date_Enroll** về dạng Datatime theo format **yyyymmdd**

In [11]:
df['Date_Enroll'] = pd.to_datetime(df['Date_Enroll'])
df['Date_Enroll'].head(5)

0   2014-06-16
1   2014-06-15
2   2014-05-13
3   2014-05-11
4   2014-04-08
Name: Date_Enroll, dtype: datetime64[ns]

## Format value type

In [12]:
# Chuyển đổi kiểu dữ liệu cột income
df["Income"] = df["Income"].str.replace("$","").str.replace(',', '')
df["Income"] = df["Income"].astype(float)

C:\Users\NGHOAN~1\AppData\Local\Temp/ipykernel_8220/3242172660.py:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df["Income"] = df["Income"].str.replace("$","").str.replace(',', '')


In [13]:
df['Income'].head(5)

0    84835.0
1    57091.0
2    67267.0
3    32474.0
4    21474.0
Name: Income, dtype: float64

In [14]:
# Chuyển đổi dữ liệu các cột chi phí
df["MntWines"] = df["MntWines"].astype(float)
df["MntFruits"] = df["MntFruits"].astype(float)
df["MntMeats"] = df["MntMeats"].astype(float)
df["MntFishs"] = df["MntFishs"].astype(float)
df["MntSweets"] = df["MntSweets"].astype(float)
df["MntGolds"] = df["MntGolds"].astype(float)

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 28 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   User_ID              2240 non-null   int64         
 1   Year_Birth           2240 non-null   int64         
 2   Education            2240 non-null   object        
 3   Marital_Status       2240 non-null   object        
 4   Income               2216 non-null   float64       
 5   Kidhome              2240 non-null   int64         
 6   Teenhome             2240 non-null   int64         
 7   Date_Enroll          2240 non-null   datetime64[ns]
 8   Recency              2240 non-null   int64         
 9   MntWines             2240 non-null   float64       
 10  MntFruits            2240 non-null   float64       
 11  MntMeats             2240 non-null   float64       
 12  MntFishs             2240 non-null   float64       
 13  MntSweets            2240 non-nul

## Result
Dữ liệu sau khi được xử lý bao gồm 38 cột và 2216 dòng

In [16]:
df.head(5)

User_ID  Year_Birth   Education Marital_Status   Income  Kidhome  Teenhome  \
0     1826        1970  Graduation       Divorced  84835.0        0         0   
1        1        1961  Graduation         Single  57091.0        0         0   
2    10476        1958  Graduation        Married  67267.0        0         1   
3     1386        1967  Graduation       Together  32474.0        1         1   
4     5371        1989  Graduation         Single  21474.0        1         0   

  Date_Enroll  Recency  MntWines  ...  NumStorePurchases  NumWebVisitsMonth  \
0  2014-06-16        0     189.0  ...                  6                  1   
1  2014-06-15        0     464.0  ...                  7                  5   
2  2014-05-13        0     134.0  ...                  5                  2   
3  2014-05-11        0      10.0  ...                  2                  7   
4  2014-04-08        0       6.0  ...                  2                  7   

   AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  \
0             0             0             0             0             0   
1             0             0             0             0             1   
2             0             0             0             0             0   
3             0             0             0             0             0   
4             1             0             0             0             0   

   Response  Complain        Country  
0         1         0          Spain  
1         1         0         Canada  
2         0         0  United States  
3         0         0      Australia  
4         1         0          Spain  

[5 rows x 28 columns]

In [17]:
# lưu vào file csv
df.to_csv('dataset/marketing_data_final.csv', index=False)